pandas/core/indexes/datetimelike.py

```diff
@@ -496,7 +496,7 @@ def _format_attrs(self):
             if attrib == "freq":
                 freq = self.freqstr
                 if freq is not None:
-                    freq = "'%s'" % freq
+                    freq = f"'{freq}'"
```

Comment:
do the single quotes become unnecessary with a `{freq:r}` or something like that?  This is a pretty common pattern, but we aren't very consistent about when we do or dont use the quotes (i sometimes use ticks like for markdown)

Reply:
Assuming you mean {freq!r} here - updated. Agree doesn't seem to be much consistency